src/main/scala/is/hail/rvd/OrderedRVDPartitioner.scala

```diff
@@ -130,8 +128,8 @@ class OrderedRVDPartitioner(
       return this
     if (range.isEmpty)
       return copy(rangeBounds = FastIndexedSeq(newRange.get))
-    val newStart = pkType.ordering.min(range.get.start, Annotation.copy(pkType, newRange.get.start))
-    val newEnd = pkType.ordering.max(range.get.end, Annotation.copy(pkType, newRange.get.end))
```

Comment:
So two things stick out to me. `enlargeToRange` is only every called on a partitioner's `range`, so this relaxation is safe as long as partitioner's range is always safe, which boils down to `rangeBounds` always being safe.

This is kind of an uncomfortable amount of assumptions and digging that we have to ensure is still correct with every new PR. I'm not suggesting we fix that in _this_ PR. Can you share a bit of the motivation for removing this particular copy?

Reply:
also related to subtyping above -- `newRange.get.start` could be a subtype of `pkType` and we're not handling that.